pyomo/contrib/alternative_solutions/lp_enum_solnpool.py

```diff
@@ -0,0 +1,234 @@
+                for var in continuous_var:
+                    if continuous_var[var].value > self.zero_threshold:
+                        num_non_zero += 1
+                        non_zero_basic_expr += binary_var[var]
+            # TODO: JLG - If we want to add the mixed binary case, I think we
+            # need to do it here. Essentially we would want to continue to
+            # build up the num_non_zero as follows
+            # for binary in binary_vars:
+            # if binary.value > 0.5:
+            # num_non_zero += 1 - binary
+            # else:
+            # num_non_zero += binary
+            new_con = self.model.cl.add(non_zero_basic_expr <= num_non_zero)
+            cb_opt.cbLazy(new_con)
+
+
+def enumerate_linear_solutions_soln_pool(
+    model,
+    num_solutions=10,
+    variables="all",
+    rel_opt_gap=None,
+    abs_opt_gap=None,
+    zero_threshold=1e-5,
+    solver_options={},
+    tee=False,
+):
+    """
+    Finds alternative optimal solutions for a (mixed-binary) linear program
+    using Gurobi's solution pool feature.
+
+        Parameters
+        ----------
+        model : ConcreteModel
+            A concrete Pyomo model
+        num_solutions : int
+            The maximum number of solutions to generate.
+        variables: 'all' or a collection of Pyomo _GeneralVarData variables
+            The variables for which bounds will be generated. 'all' indicates
+            that all variables will be included. Alternatively, a collection of
+            _GenereralVarData variables can be provided.
+        rel_opt_gap : float or None
+            The relative optimality gap for the original objective for which
+            variable bounds will be found. None indicates that a relative gap
+            constraint will not be added to the model.
+        abs_opt_gap : float or None
+            The absolute optimality gap for the original objective for which
+            variable bounds will be found. None indicates that an absolute gap
+            constraint will not be added to the model.
+        zero_threshold: float
+            The threshold for which a continuous variables' value is considered
+            to be equal to zero.
+        solver_options : dict
+            Solver option-value pairs to be passed to the solver.
+        tee : boolean
+            Boolean indicating that the solver output should be displayed.
+
+        Returns
+        -------
+        solutions
+            A list of Solution objects.
+            [Solution]
+    """
+    print("STARTING LP ENUMERATION ANALYSIS USING GUROBI SOLUTION POOL")
+    #
+    # Setup gurobi
+    #
+    if not gurobi_available:
+        return []
+
+    # For now keeping things simple
+    # TODO: See if this can be relaxed, but for now just leave as all
+    assert variables == "all"
+    if variables == "all":
+        all_variables = aos_utils.get_model_variables(model, "all")
+
+    # TODO: Check if problem is continuous or mixed binary
```

Comment:
@jlgearh ???

Reply:
This needs to throw an error if there are binaries. It's only implemented for continuous now.

Reply:
Done

Reply:
Unresolved this for the moment because I talked with @jlgearh and this needs to check that no variables are discrete to be correct--he plans to make the change.

Reply:
I updated this to check for any discrete variables.